packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py

```diff
@@ -284,6 +306,45 @@ async def upload_path(
     :raises exceptions.NodeportsException
     :return: stored id, S3 entity_tag
     """
+    async for attempt in AsyncRetrying(
+        reraise=True,
+        wait=wait_random_exponential(),
+        stop=stop_after_attempt(
+            NodePortsSettings.create_from_envs().NODE_PORTS_400_REQUEST_TIMEOUT_ATTEMPTS

```

Comment:
I thought we finally got rid of this way of getting settings! :-(((

Reply:
in `simcode-sdk` and `nodeports` has them like this.
A bit inconvenient since you don't have an "app" to which they are attached and from where you can recover them easily.

Reply:
yes. I am aware of this. The problem I see is that we are not only not refactoring but making the problem bigger. Just a though